### PR TITLE
3.7.3-wechatprogram merge to 3.7.3

### PR DESCRIPTION
**Pull request by novain1102**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cc.config.json [hunk 1]
```diff
@@ -370,6 +370,13 @@
             "ccGlobal": true,
             "internal": false
         },
+        "WECHAT_MINI_PROGRAM": {
+            "comment": "Running in the Wechat's mini program.",
+            "type": "boolean",
+            "value": false,
+            "ccGlobal": true,
```
Comment (PPpro) [inline cc.config.json:377]: remove this item, it's a legacy option for Cocos 2.x

File: pal/system-info/minigame/system-info.ts [hunk 2]
```diff
@@ -125,13 +127,13 @@ class SystemInfo extends EventTarget {
 
         this.isXR = false;
 
-        const isPCWechat = WECHAT && this.os === OS.WINDOWS && !minigame.isDevTool;
+        const isPCWechat = (WECHAT || WECHAT_MINI_PROGRAM) && this.os === OS.WINDOWS && !minigame.isDevTool;
```
Comment (PPpro) [inline pal/system-info/minigame/system-info.ts:130]: we don't support PC wechat mini program, so please remove this condition

File: pal/system-info/minigame/system-info.ts [hunk 3]
```diff
@@ -125,13 +127,13 @@ class SystemInfo extends EventTarget {
...
         this._featureMap = {
             [Feature.WEBP]: false,      // Initialize in Promise,
             [Feature.IMAGE_BITMAP]: false,
             [Feature.WEB_VIEW]: false,
-            [Feature.VIDEO_PLAYER]: WECHAT || OPPO,
-            [Feature.SAFE_AREA]: WECHAT || BYTEDANCE,
+            [Feature.VIDEO_PLAYER]: WECHAT || WECHAT_MINI_PROGRAM || OPPO,
+            [Feature.SAFE_AREA]: WECHAT || WECHAT_MINI_PROGRAM || BYTEDANCE,
```
Comment (PPpro) [inline pal/system-info/minigame/system-info.ts:136]: does wechat mini program support video and safe area ?

Comment (novain1102) [reply pal/system-info/minigame/system-info.ts:136]: Yes. But need to adapte in future.

File: pal/system-info/minigame/system-info.ts (hunk 2 above)
Comment (novain1102) [reply pal/system-info/minigame/system-info.ts:130]: ok.

**Pull request closed without merging**